sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala

```diff
@@ -254,6 +254,11 @@ object HiveThriftServer2 extends Logging {
       trimExecutionIfNecessary()
     }
 
+    def onOperationClosed(id: String): Unit = synchronized {
+      executionList(id).finishTimestamp = System.currentTimeMillis
```

Comment:
I would add a separate field `closedTimestamp`, and add a separate column in the tables in the UI (overall and within session). What do you think?

Reply:
+1

Reply:
How about?
![image](https://user-images.githubusercontent.com/5399861/60803522-ff879300-a1ad-11e9-8c2c-a646de0f06e0.png)
`Execution Time` = `Finish Time` - `Start Time`
`Duration` = `Close Time` - `Start Time`

Reply:
or just add `Fetch result time` (Fetch result time = Close Time - Finish Time):

Start Time | Finish Time | Duration | Fetch result time | Statement | State
-- | -- | -- | -- | -- | --

Reply:
I would vote for the former - add `Close time` and `Execution time` and `Duration`.
`Fetch result time` is a bit of a long label, and also I think it might be slightly misleading - this may also be the client being slow in processing the results, just sitting on an idle cursor, or actually closing it without fetching all results etc.
Having `Duration` for the total time gives all the same information and I think it's also more relevant, to have the time from start to close.